extensions/wikia/PageHeader/PageTitle.class.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Wikia\PageHeader;
+
+use \RequestContext;
+use \Title;
+use \WikiaGlobalRegistry;
+use \WikiaApp;
+use \WikiaPageType;
+
+class PageTitle {
+
+	/* @var string */
+	public $title;
+
+	/* @var string */
+	public $prefix;
+
+	/* @var WikiaGlobalRegistry */
+	private $wg;
+
+	/* @var int */
+	private $namespace;
+
+	/* @var Title */
+	private $MWTitle;
+
+	const PREFIX_LESS_NAMESPACES = [ NS_MEDIAWIKI, NS_TEMPLATE, NS_CATEGORY, NS_FILE ];
+
+	public function __construct( WikiaApp $app ) {
+		$this->wg = $app->wg;
+		$this->MWTitle = RequestContext::getMain()->getTitle();
+		$this->request = RequestContext::getMain()->getRequest();
+
+		$this->namespace = $this->MWTitle->getNamespace();
+		$this->title = $this->handleTitle( $app );
+		$this->prefix = $this->handlePrefix();
+	}
+
+	private function handleTitle( WikiaApp $app ): string {
+		if ( WikiaPageType::isMainPage() ) {
+			return $this->titleMainPage();
+		} else if ( $this->MWTitle->isTalkPage() || $this->shouldNotDisplayNamespacePrefix( $this->namespace ) ) {
+			return htmlspecialchars( $this->MWTitle->getText() );
+		} else if ( $this->request->getCheck( 'wpPreview' ) || $this->request->getCheck( 'wpLivePreview' ) ) {
+			return $this->prefixedTitle( 'oasis-page-header-preview' );
+		} else if ( $this->request->getCheck( 'wpDiff' ) ) {
+			return $this->prefixedTitle( 'oasis-page-header-changes' );
+		} else if ( $this->request->getCheck( 'diff' ) ) {
+			return $this->prefixedTitle( 'oasis-page-header-diff' );
+		} else if ( is_numeric( $this->request->getVal( 'section' ) ) ) {
+			return $this->prefixedTitle( 'oasis-page-header-editing-section' );
```

Comment:
I think we can remove this. The only way I managed to display it was to open http://starwars.wikia.com/wiki/Yoda?section=1 and it shouldn't be visible there anyway. On http://starwars.wikia.com/wiki/Yoda?section=1&action=edit we use a different message - `editingsection`